autonomic-plugin-common/src/test/java/br/com/autonomiccs/autonomic/plugin/common/services/AutonomiccsServiceOfferingServiceTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * This program is part of Autonomiccs "autonomic-platform",
+ * an open source autonomic cloud computing management platform.
+ * Copyright (C) 2016 Autonomiccs, Inc.
+ *
+ * Licensed to the Autonomiccs, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The Autonomiccs, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package br.com.autonomiccs.autonomic.plugin.common.services;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import com.cloud.service.ServiceOfferingVO;
+import com.cloud.service.dao.ServiceOfferingDao;
+import com.cloud.storage.Storage;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.cloud.vm.VirtualMachine;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AutonomiccsServiceOfferingServiceTest {
+
+    @Spy
+    @InjectMocks
+    private AutonomiccsServiceOfferingService spy;
+    @Mock
+    private ServiceOfferingDao serviceOfferingDao;
+
+    private static final int AUTONOMICCS_SYSTEM_VM_VM_RAMSIZE = 512;
+    private static final int AUTONOMICCS_SYSTEM_VM_CPUMHZ = 1000;
+    private static final String AUTONOMICCS_SYSTEM_VM_SERVICE_OFFERING_UNIQUE_NAME = "Autonomiccs S.O.";
+    private static final String AUTONOMICCS_SYSTEM_VM_SERVICE_OFFERING_NAME = "Autonomiccs system VMs service offering";
+
+    @Test
+    public void searchServiceOfferingByNameTest() {
+        List<ServiceOfferingVO> allServiceOffering = createServiceOfferingVOList(1);
+        Mockito.doReturn(allServiceOffering).when(serviceOfferingDao).listAll();
+
+        ServiceOfferingVO result = spy.searchServiceOfferingByName("serviceOfferingVoName");
+
+        Mockito.verify(serviceOfferingDao).listAll();
+        Assert.assertEquals(allServiceOffering.get(0), result);
+    }
+
+    @Test(expected = CloudRuntimeException.class)
+    public void searchServiceOfferingByNameTestExpectCloudRuntimeException() {
```

Comment:
I can't figure out exactly _when_ the `searchServiceOfferingByName` method would throw that exception. Looking at the code, that method would simply return a null reference when it can't match the name passed by parameter to an existing one...